frontend/js/src/common/listens/ListenCard.tsx

```diff
@@ -761,5 +742,40 @@ export class ListenCard extends React.Component<
 
 export default function ListenCardWrapper(props: ListenCardProps) {
   const dispatch = useBrainzPlayerDispatch();
-  return <ListenCard {...props} dispatch={dispatch} />;
+  const { spotifyAuth, APIService, userPreferences } = React.useContext(
+    GlobalAppContext
+  );
+  const { listen, customThumbnail } = props;
+
+  const albumArtQueryKey = React.useMemo(
+    () => getAlbumArtFromListenMetadataKey(listen, spotifyAuth),
+    [listen, spotifyAuth]
+  );
+
+  const albumArtDisabled =
+    Boolean(customThumbnail) || !listen || userPreferences?.saveData;
+
+  const { data: thumbnailSrc } = useQuery({
+    queryKey: ["album-art", albumArtQueryKey, albumArtDisabled],
+    queryFn: async () => {
+      if (albumArtDisabled) {
+        return undefined;
+      }
+      try {
+        return await getAlbumArtFromListenMetadata(
+          listen,
+          spotifyAuth,
+          APIService
+        );
+      } catch (error) {
+        return undefined;
```

Comment:
Might be useful to print out the rare errors to the console, in case users are having issues or don't understand why cover art doesn't load.
For example a user previously reported issues that were caused by their browser blocking the spotify API calls.

Reply:
Fixed